.eslintrc.js

```diff
@@ -1,228 +1,182 @@
-    'prefer-template': 'error',
-    'require-yield': 'error',
-
-    // React
-    'react/display-name': 'off',
-    'react/forbid-prop-types': 'off',
-    'react/no-comment-textnodes': 'off',
-    'react/no-danger': 'error',
-    'react/no-deprecated': 'error',
-    'react/no-did-mount-set-state': 'error',
-    'react/no-did-update-set-state': 'error',
-    'react/no-direct-mutation-state': 'error',
-    'react/no-is-mounted': 'error',
-    'react/no-multi-comp': ['error', { ignoreStateless: true }],
-    'react/no-render-return-value': 'error',
-    'react/no-set-state': 'off',
-    'react/no-string-refs': 'error',
-    'react/no-unknown-property': 'error',
-    'react/prefer-es6-class': ['error', 'always'],
-    'react/prefer-stateless-function': 'off', // 'error',
-    'react/prop-types': ['error', { skipUndeclared: true }],
-    'react/react-in-jsx-scope': 'error',
-    'react/require-extension': 'off',
-    'react/require-optimization': 'off',
-    'react/require-render-return': 'error',
-    'react/self-closing-comp': 'error',
-    'react/sort-comp': 'error',
-    'react/sort-prop-types': 'off',
-    'react/jsx-boolean-value': 'off',
-    'react/jsx-curly-spacing': 'off', // [2, "never", { "allowMultiline": false }],
-    'react/jsx-filename-extension': ['error', { extensions: ['.js'] }],
-    'react/jsx-handler-names': 'off', // 'error',
-    'react/jsx-key': 'error',
-    'react/jsx-max-props-per-line': 'off',
-    'react/jsx-no-bind': 'off',
-    'react/jsx-no-duplicate-props': 'error',
-    'react/jsx-no-literals': 'off',
-    'react/jsx-no-target-blank': 'error',
-    'react/jsx-no-undef': 'error',
-    'react/jsx-pascal-case': 'error',
-    'react/jsx-sort-props': 'off',
-    'react/jsx-uses-react': 'error',
-    'react/jsx-uses-vars': 'error',
-    'react-hooks/rules-of-hooks': 'error',
-    'react-hooks/exhaustive-deps': 'error',
-
-    // import errors
-    'import/no-unresolved': [
-      'error',
-      {
-        commonjs: true,
-        amd: true,
-        ignore: ['.svg$', '^file?']
-      }
-    ],
-    'import/named': 'error',
-    'import/namespace': 'error',
-    'import/default': 'error',
-    'import/export': 'error'
+      version: '>16',
+    },
   },
-  plugins: ['react', 'react-hooks', 'css-modules', 'import', 'flowtype', 'cypress', 'jest'],
-  extends: [
-    'plugin:jest/recommended',
-    'plugin:css-modules/recommended',
-    'prettier',
-    'plugin:flowtype/recommended',
-    'plugin:cypress/recommended'
-  ]
+  rules,
+  plugins: [
+    'react',
+    'react-hooks',
+    'css-modules',
+    'cypress',
+    'jest',
+  ],
+  extends: commonExtends,
+  overrides: [
+    {
+      files: ['**/*.ts', '**/*.tsx'],
+      parser: '@typescript-eslint/parser',
+      settings: {
+        'import/parsers': {
+          'babel-eslint': ['.js'],
+        },
+      },
+      extends: [
+        'plugin:@typescript-eslint/recommended',
+        'prettier/@typescript-eslint',
+        ...commonExtends,
```

Comment:
Is it necessary to spread the common `extends` and `rules` in the `overrides`? Only asking because we dont seem to be doing that in the js overrides.